api/src/main/java/org/commonjava/indy/data/StoreValidatorRemote.java

```diff
@@ -0,0 +1,162 @@
+package org.commonjava.indy.data;
+
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpHead;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.commonjava.indy.model.core.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.*;
+
+
+@RemoteStoreValidator
+public class StoreValidatorRemote implements StoreValidator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StoreValidatorRemote.class);
+    private final CountDownLatch httpRequestsLatch = new CountDownLatch(2);
+    private final ExecutorService executorService = Executors.newFixedThreadPool(2);
```

Comment:
We should use the Weft annotations to CDI inject this field, so we gain the ability to configure threadpool size, priority, etc. and also expose it for APM measurement + eventual overflow / job-rejection quotas.

Those annotations look like this:
```
    @Inject
    @WeftManaged
    @ExecutorConfig( name="store-validation", threads=10, priority=6 )
    private ExecutorService executorService;
```